tests/test_sklearn_linear_converters.py

```diff
@@ -0,0 +1,158 @@
+"""
+Tests sklearn linear classifiers (LinearRegression, LogisticRegression, SGDClassifier, LogisticRegressionCV) converters.
+"""
+import unittest
+import warnings
+
+import numpy as np
+import torch
+from sklearn.linear_model import LinearRegression, LogisticRegression, SGDClassifier, LogisticRegressionCV
+
+import hummingbird.ml
+
+
+class TestSklearnLinearClassifiers(unittest.TestCase):
+
+    # LogisticRegression test function to be parameterized
+    def _test_logistic_regression(self, num_classes, solver="liblinear", multi_class="auto", labels_shift=0):
+        if num_classes > 2:
+            model = LogisticRegression(solver=solver, multi_class=multi_class, fit_intercept=True)
+        else:
+            model = LogisticRegression(solver="liblinear", fit_intercept=True)
+
+        np.random.seed(0)
+        X = np.random.rand(100, 200)
+        X = np.array(X, dtype=np.float32)
+        y = np.random.randint(num_classes, size=100) + labels_shift
+
+        model.fit(X, y)
+
+        pytorch_model = hummingbird.ml.convert(model, "pytorch")
+
+        self.assertTrue(pytorch_model is not None)
+        np.testing.assert_allclose(model.predict_proba(X), pytorch_model.predict_proba(X), rtol=1e-6, atol=1e-6)
+
+    # LogisticRegression binary
+    def test_logistic_regression_bi(self):
+        self._test_logistic_regression(2)
+
+    # LogisticRegression multiclass with auto
+    def test_logistic_regression_multi_auto(self):
+        self._test_logistic_regression(3)
+
+    # LogisticRegression with class labels shifted
+    def test_logistic_regression_shifted_classes(self):
+        self._test_logistic_regression(3, labels_shift=2)
+
+    # LogisticRegression with multi+ovr
+    def test_logistic_regression_multi_ovr(self):
+        self._test_logistic_regression(3, multi_class="ovr")
+
+    # LogisticRegression with multi+multinomial
+    def test_logistic_regression_multi_multin(self):
+        warnings.filterwarnings("ignore")
+        # this will not converge due to small test size
+        self._test_logistic_regression(3, multi_class="multinomial", solver="sag")
+
+    # LinearRegression test function to be parameterized
+    def _test_linear_regression(self, y_input):
```

Comment:
We should do the same also for the other regression tests.

Reply:
Which exactly? these (at least not all of these) tests don't make sense for LinearReg...  Coverage is complete; is there a specific sub-case you have in mind that is missing?  Or did you mean for SDG?